docling/datamodel/pipeline_options.py

```diff
@@ -29,6 +29,42 @@ class OcrOptions(BaseModel):
     )
 
 
+class RapidOcrOptions(OcrOptions):
```

Comment:
@cau-git do you think we should simply expose the raw parameters or trying to simplify them?

@Swaymaw what is your understanding about supporting cpu-only, mac mps, cuda, (others) devices?
Is there a simple way to switch among those?
We would like to have this clearly documented for the new release with device support we are preparing.

Reply:
We can detect devices using torch or other backends in an automated fashion to switch to the fastest device for the user, in that case we would have to set a priority or something if more than one device is available. However, I personally prefer more of a PyTorch style approach where we explicitly mention the device we want to use, it makes debugging easier and also makes the code more device agnostic at the same time it makes the code much more readable.

Reply:
<img width="406" alt="Screenshot 2024-11-26 at 5 17 47 PM" src="https://github.com/user-attachments/assets/154956e3-98b4-4ba2-9b0a-3124f4b93c72">

[Source](https://github.com/RapidAI/RapidOCR/blob/c7b1a6000fa7facc00d49c83888f598c4de0df7c/python/rapidocr_onnxruntime/utils/infer_engine.py#L25)

RapidOCR does not have a support for MPS devices. 
<hr>
However, to use CPU, CUDA or DirectML Devices you can use the following options: 

- For CPU-Only: (Default- Not Providing any options defaults to CPU-Only)
  ```python
  RapidOcrOptions(
         det_use_cuda = False
         cls_use_cuda = False
         rec_use_cuda = False 

         det_use_dml = False  
         cls_use_dml = False  
         rec_use_dml = False  
     ) 
  ```

- For cuda: 
    ```python
    RapidOcrOptions(
      det_use_cuda = True
      cls_use_cuda = True
      rec_use_cuda = True 
  ) 
    ```
- For DirectML: 
    ```python
    RapidOcrOptions(
         det_use_dml = True
         cls_use_dml = True  
         rec_use_dml = True  
  ) 
    ```